File: aoa_server/communication/MessageType.cpp
#include <iostream>
#include "MessageType.h"

MessageType convertInternalMessageType(int index) {
    switch(index % MESSAGE_TYPE_COUNT) {
        case 0: return MessageType::HELLO;
        case 1: return MessageType::SIGN_IN;
        case 2: return MessageType::GAME_LIST;
        case 3: return MessageType::GAME_NEW;
        case 4: return MessageType::GAME_JOIN;
        case 5: return MessageType::GAME_START;
        case 6: return MessageType::TURN_DATA;
        case 7: return MessageType::GAME_END;
        case 8: return MessageType::GAME_LEAVE;
        case 9: return MessageType::SIGN_OUT;
        case 10: return MessageType::PLAYER_INFO;
        default: case 11: return MessageType::WAIT_READY;
    }
};

File: aoa_client/src/cz/kiv/ups/communication/MessageType.java
package cz.kiv.ups.communication;

import cz.kiv.ups.config.CommunicationConfig;

public enum MessageType {
    HELLO (CommunicationConfig.REQ_HELLO),
    SIGN_IN (CommunicationConfig.REQ_SIGN_IN),
    GAME_LIST (CommunicationConfig.REQ_GAME_LIST),
    GAME_NEW (CommunicationConfig.REQ_GAME_NEW),
    GAME_JOIN (CommunicationConfig.REQ_GAME_JOIN),
    GAME_START (CommunicationConfig.REQ_GAME_START),
    TURN_DATA (CommunicationConfig.REQ_TURN_DATA),
    GAME_RESULT(CommunicationConfig.REQ_GAME_END),
    GAME_LEAVE (CommunicationConfig.REQ_GAME_END),
    SIGN_OUT (CommunicationConfig.REQ_SIGN_OUT),
    PLAYER_LIST (CommunicationConfig.REQ_PLAYER_LIST);

    private int code;

    MessageType(int code) {
        this.code = code;
    }

    public int getCode() {
        return code;
    }

    public static MessageType nth(int index){
        for (MessageType t : MessageType.values()) {
            if(index == t.getCode()) {
                return t;
            }
        }
        return null;
    }
}
